clang/lib/Tooling/DependencyScanning/DependencyScanningFilesystem.cpp

```diff
@@ -108,6 +108,32 @@ DependencyScanningFilesystemSharedCache::getShardForUID(
   return CacheShards[Hash % NumShards];
 }
 
+void DependencyScanningFilesystemSharedCache::diagnoseNegativeStatCachedPaths(
+    llvm::raw_ostream &OS, llvm::vfs::FileSystem &UnderlyingFS) const {
+  // Iterate through all shards and look for cached stat errors.
+  for (unsigned i = 0; i < NumShards; i++) {
+    const CacheShard &Shard = CacheShards[i];
+    for (const auto &Path : Shard.CacheByFilename.keys()) {
+      auto CachedPairIt = Shard.CacheByFilename.find(Path);
```

Comment:
Why the extra lookup?
```suggestion
    for (const auto &[Filename, Entry] : Shard.CacheByFilename) {
      const CachedFileSystemEntry *Entry = Entry.first;
```

Reply:
Thanks! Fixed.